quadtree.js

```diff
@@ -82,9 +82,15 @@ class Circle {
 }
 
 class QuadTree {
-  constructor(boundary, n) {
+  constructor(boundary, capacity) {
+    if (!boundary) {
+      throw TypeError('boundary is null or undefined');
+    }
+    if (capacity < 1) {
+      throw RangeError('capacity must be greater than 0');
+    }
```

Comment:
These checks are not enough IMO: checking the existence of Boundary still means we can set pears or apples.

To be ideally checked, boundary should be validated as a real boundary element 😄

Reply:
Same for "Capacity" by the way

Reply:
Made these validations more specific, added unit tests for the changes